frontend/device/generic/device.lua

```diff
@@ -920,8 +921,12 @@ function Device:unpackArchive(archive, extract_to)
     return true
 end
 
-function Device:untar(archive, extract_to)
-    return os.execute(("./tar xf %q -C %q"):format(archive, extract_to))
+function Device:untar(archive, extract_to, with_stripped_root)
```

Comment:
This function is also in https://github.com/koreader/koreader/blob/master/frontend/device/android/device.lua#L514.

Not sure what '--strip-components=1' does but we need to check how android behaves without any additional change with this PR in and/or change the untar logic in https://github.com/koreader/android-luajit-launcher/blob/master/app/src/main/java/org/koreader/launcher/extensions/FileExtensions.kt#L74-L126

Reply:
Or leave this function signature as is and do "strip components" by hand after the tar file is uncompressed

Reply:
I don't think it will cause any issues, the `--strip-components=1` will remove the root directory in the archive (if it's present).
So when we pass in the `extract_to` <data_dir>/<dict.name> directory it doesn't create an unnecessary nested folder inside that deepest directory. 

On android there would be an unnecessary nested directory, but it wouldn't fail right? It looks like it'll make the `extract_dir` just fine, I assume it can create parent directories fine as well?

Reply:
Thanks for the info.

> On android there would be an unnecessary nested directory, but it wouldn't fail right? 

Nope, it should work fine. Still I would prefeer to keep the same expected behaviour but I think it doesn't need to be addressed in this PR.


>  the --strip-components=1 will remove the root directory in the archive (if it's present).

Do you have some samples of dicts with and without an inner folder?

Reply:
I couldn't immediately find one so I made one myself for a random small one of the list: https://github.com/Vuizur/Wiktionary-Dictionaries/raw/master/Alemannic%20German-English%20Wiktionary%20dictionary%20stardict.tar.gz
```sh
$ tar tf Alemannic%20German-English%20Wiktionary%20dictionary%20stardict.tar.gz 
Alemannic German-English Wiktionary dictionary stardict/
Alemannic German-English Wiktionary dictionary stardict/Alemannic German-English Wiktionary dictionary.dict.dz
Alemannic German-English Wiktionary dictionary stardict/Alemannic German-English Wiktionary dictionary.idx
Alemannic German-English Wiktionary dictionary stardict/Alemannic German-English Wiktionary dictionary.ifo
Alemannic German-English Wiktionary dictionary stardict/Alemannic German-English Wiktionary dictionary.syn
$ # this one clearly has a root directory
$ tar cfz german-english-noroot.tar.gz -C Alemannic\ German-English\ Wiktionary/ 'Alemannic German-English Wiktionary dictionary.'{dict.dz,ifo,idx,syn}
$ tar tf german-english-noroot.tar.gz 
Alemannic German-English Wiktionary dictionary.dict.dz
Alemannic German-English Wiktionary dictionary.ifo
Alemannic German-English Wiktionary dictionary.idx
Alemannic German-English Wiktionary dictionary.syn
$ # root's gone!
```
[german-english-noroot.tar.gz](https://github.com/koreader/koreader/files/10931594/german-english-noroot.tar.gz)